json-endpoints/jetpack/class.jetpack-json-api-plugins-install-endpoint.php

```diff
@@ -78,12 +77,20 @@ protected static function get_plugin_id_by_slug( $slug ) {
 			return false;
 		}
 		foreach( $plugins as $id => $plugin_data ) {
-			if ( strpos( $id, $slug ) !== false ) {
+			if ( self::get_slug_from_id( $id ) === $slug ) {
 				return $id;
 			}
 		}
 		return false;
 	}
+
+	protected static function get_slug_from_id( $id ) {
+		$dir = dirname ( $id );
+		if ( $dir === '.' ) {
```

Comment:
`dirname( 'file.php' )` will output `file.php`, so it will not likely every be equal to '.'
So we could use `strpos` to find out if the plugin is in a directory, or a regex like
```
$dir = dirname( $id );
return preg_match( '/^[a-zA-Z0-9_\-]*\/[a-zA-Z0-9_\-]*\.php/', $dir ) ? $dir : substr( $id, 0, -4 );
```